corehq/blobs/metadata.py

```diff
@@ -31,6 +32,8 @@ def new(self, **blob_meta_args):
                     "keyword arguments are incompatible with `meta` argument")
             return blob_meta_args["meta"]
         timeout = blob_meta_args.pop("timeout", None)
+        if blob_meta_args.get('type_code') == CODES.form_xml:
+            blob_meta_args['compressed_length'] = -1
```

Comment:
I don't really like this. It does got overwritten during later but still feels off